Makefile

```diff
@@ -237,6 +237,7 @@ CMAKE_FILES+= \
 endif
 
 #----------------------------------------------------------------------------------------------
+BOOST_DIR ?= $(ROOT)/.install/boost
```

Comment:
I think this is ideal for CI but not for local development. IMO we should keep the default empty value, especially as this is a non-existing directory and is not ignored by `.gitignore`

Reply:
Then passing BOOST_DIR is mandatory...
I think it's a good practice to have a default value, allowing you to just run the script and go.
And we can add .install/boost_* to .gitignore :)